### PR TITLE
refactor: move errors from operation plan

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/jit/exec.rs [hunk 1]
```diff
@@ -6,19 +6,38 @@ use derive_getters::Getters;
...
+        self.err.lock().unwrap().clone()
+    }
+}
+
 ///
 /// Default GraphQL executor that takes in a GraphQL Request and produces a
 /// GraphQL Response
 pub struct Executor<IRExec, Input> {
-    plan: OperationPlan<Input>,
+    env: ExecutionEnv<Input>,
```
Comment (tusharmath) [inline src/core/jit/exec.rs:40]: Executor should be created once and not per request, hence this mutable state should move to context.

Comment (meskill) [reply src/core/jit/exec.rs:40]: actually the executor is created per request. The executor uses OperationPlan and OperationPlan is created based on request

File: src/core/jit/exec.rs [hunk 2]
```diff
@@ -6,19 +6,38 @@ use derive_getters::Getters;
...
 use crate::core::jit::synth::Synth;
+use crate::core::jit::{self};
 use crate::core::json::JsonLike;
 
 type SharedStore<Output, Error> = Arc<Mutex<Store<Result<Output, Error>>>>;
 
+#[derive(Debug, Clone, Getters)]
+pub struct ExecutionEnv<Input> {
+    plan: OperationPlan<Input>,
+    err: Arc<Mutex<Vec<LocationError<jit::error::Error>>>>,
```
Comment (meskill) [inline src/core/jit/exec.rs:20]:
```suggestion
    errors: Arc<Mutex<Vec<LocationError<jit::error::Error>>>>,
```

File: src/core/jit/exec.rs [hunk 3]
```diff
@@ -6,19 +6,38 @@ use derive_getters::Getters;
...
+impl<Input> ExecutionEnv<Input> {
+    pub fn new(plan: OperationPlan<Input>) -> Self {
+        Self { plan, err: Arc::new(Mutex::new(vec![])) }
+    }
+    pub fn add_error(&self, new_error: LocationError<jit::error::Error>) {
+        self.err.lock().unwrap().push(new_error);
+    }
+
+    pub fn get_errors(&self) -> Vec<LocationError<jit::error::Error>> {
+        self.err.lock().unwrap().clone()
```
Comment (meskill) [inline src/core/jit/exec.rs:32]: don't do clone here, better let caller decide what to do

**Review by meskill — CHANGES_REQUESTED** [review]

File: src/core/jit/exec.rs (hunk 2 above)
Comment (laststylebender14) [reply src/core/jit/exec.rs]: done

File: src/core/jit/exec.rs (hunk 3 above)
Comment (laststylebender14) [reply src/core/jit/exec.rs]: done

File: src/core/jit/context.rs [hunk 4]
```diff
@@ -113,7 +112,9 @@ mod test {
     fn test_field() {
         let (plan, req) = setup("query {posts {id title}}");
         let field = plan.as_nested();
-        let ctx: Context<ConstValue, ConstValue> = Context::new(&req, false, &field[0]);
-        insta::assert_debug_snapshot!(<Context<_, _> as ResolverContextLike>::field(&ctx).unwrap());
+        let env = ExecutionEnv::new(plan.clone());
```
Comment (laststylebender14) [inline src/core/jit/context.rs:115]: consume plan without cloning

**Review by meskill — CHANGES_REQUESTED** [review]

File: src/core/jit/context.rs [hunk 5]
```diff
@@ -113,7 +112,9 @@ mod test {
...
+        let ctx: Context<async_graphql::Value, async_graphql::Value> =
```
[the remark this answers is not in the document]
Comment (laststylebender14) [reply src/core/jit/context.rs:116]: done

File: src/core/jit/context.rs [hunk 6]
```diff
@@ -113,7 +112,9 @@ mod test {
...
+            Context::new(&req, &field[0], &env);
+        insta::assert_debug_snapshot!(ctx.field());
```
[the remark this answers is not in the document]
Comment (laststylebender14) [reply src/core/jit/context.rs]: forgot to revert this change, thanks for pointing out.

**Pull request merged**